src/php5/solr_functions_params.c

```diff
@@ -713,10 +713,8 @@ PHP_SOLR_API void solr_arg_list_param_value_fetch(solr_param_t *solr_param, solr
 	{
 		solr_string_append_solr_string(&tmp_buffer, &(current_ptr->contents.arg_list.value));
 
-		if (current_ptr->contents.arg_list.delimiter_override) {
-		    if (strlen(current_ptr->contents.arg_list.delimiter_override) > 0) {
-		        solr_string_appendc(&tmp_buffer, *current_ptr->contents.arg_list.delimiter_override);
-		    }
+		if (current_ptr->contents.arg_list.delimiter_override && strlen(current_ptr->contents.arg_list.delimiter_override) > 0) {
```

Comment:
the length calculation is not actually needed, it's only needed to check the value is not empty. considering it is [asciiz](https://en.wikipedia.org/wiki/Null-terminated_string) string, the index=0 should not be `\0`:

```c
if (current_ptr->contents.arg_list.delimiter_override && *current_ptr->contents.arg_list.delimiter_override) {
```

or:

```c
if (current_ptr->contents.arg_list.delimiter_override && current_ptr->contents.arg_list.delimiter_override[0]) {
```

whichever you find more readable.

Reply:
seems rest of the code uses first variant.

Reply:
Hi @glensc,
OK, will do this late evening.

I'll add a new test scenario too, the one testing [`addQueryField()`](https://php.net/manual/en/solrdismaxquery.addqueryfield.php) w/ a `boost` value, the one failing using my "quick fix" instead of your "tidy up suggestion" which was more than that i.e. a fix in my broken logic 😊.

TNX for your patience, appreciated 😉 .